Enhance error handling for `composer update` in `env:start`

Adi-ty opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

This PR improves the developer experience for `npm run env:start` by providing clear, actionable error messages if `composer update` fails.

Trac ticket: https://core.trac.wordpress.org/ticket/62807

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

Adi-ty committed:
feat: Enhance error handling for composer update in env:start

azaozz commented on tools/local-env/scripts/start.js:
```diff
@@ -48,6 +48,30 @@ spawnSync(
+);
+
+if ( result.status !== 0 ) {
+	console.error( '\nThere was a failure running composer update.' );
```
Just a nitpick but perhaps this warning message can be a bit better? The first time it seemed a bit unclear as Docker also has a `compose` command. Maybe something like: "There was an error while running Composer update."? Or even maybe "There was an error while running PHP Composer update."?

azaozz committed:
Merge remote-tracking branch 'upstream/trunk' into pr/9360

azaozz commented:
Seems there's a left-over orphan Docker container when the Composer update fails. Maybe it can be removed at the end there?

Adi-ty committed:
feat: Improve error handling for PHP composer update in local environment

- Adds `--rm` flag to automatically remove containers after composer update completes, preventing orphaned containers
- Updates error message to "There was an error while running PHP Composer update" for clarity, distinguishing it from Docker Compose commands

Adi-ty commented:
> Seems there's a left-over orphan Docker container when the Composer update fails. Maybe it can be removed at the end there?

Thank you for the feedback! I've updated the PR to address both concerns

azaozz committed:
Merge branch 'enhancement/62807-env-start-composer-error-handling' of https://github.com/Adi-ty/wordpress-develop into pr/9360

azaozz reviewed (approved):
Looks good imho.

Adi-ty committed:
Merge branch 'trunk' into enhancement/62807-env-start-composer-error-handling

Adi-ty committed:
Fix: Correct syntax error in `env:start` script in package.json
